server/src/main/java/com/feeder/server/provider/twitter/TwitterCredentials.java

```diff
@@ -0,0 +1,32 @@
+package com.feeder.server.provider.twitter;
+
+import com.feeder.server.ApplicationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import twitter4j.Twitter;
+import twitter4j.TwitterFactory;
+import twitter4j.conf.ConfigurationBuilder;
+
+@Configuration
+class TwitterCredentials {
+    private TwitterFactory twitterFactory;
+    TwitterCredentials(ApplicationProperties properties) {
+        ConfigurationBuilder cb = new ConfigurationBuilder();
+        cb.setDebugEnabled(true)
+                .setOAuthConsumerKey(properties.getTwitterConsumerKey())
+                .setOAuthConsumerSecret(properties.getTwitterConsumerSecret())
+                .setOAuthAccessToken(properties.getTwitterAccessToken())
+                .setOAuthAccessTokenSecret(properties.getTwitterAccessTokenSecret());
+        twitterFactory = new TwitterFactory(cb.build());
+    }
+
+    @Bean
+    public Twitter getTwitterInstance(ApplicationProperties properties) {
+        if (twitterFactory != null) {
+            return twitterFactory.getInstance();
+        } else {
+            new TwitterCredentials(properties);
+            return twitterFactory.getInstance();
+        }
```

Comment:
What happens if this was just:

```suggestion
            return twitterFactory.getInstance();
```
Also I don't understand why `TwitterCredentials` is created then discarded.

Reply:
You need to build the factory with configurations first.. could have placed it in this method but chose to not do that because it didn't make sense to me to

Reply:
I tested it with this change and it seems to work fine?

Reply:
Weirdly yes.. will change